test/minikube/minikube_rolling_upgrade_test.go

```diff
@@ -232,6 +240,62 @@ func TestKubernetesUpgradeFullDatabaseMinorVersion(t *testing.T) {
 
 		verifyAllProcessesRunning(t, namespaceName, admin0, 2)
 	})
+
+	testlib.RunOnNuoDBVersionCondition(t, ">4.2.2", func(version *semver.Version) {
+		// check that KAA will upgrade database protocol version and restart TE
+		// automatically
+		t.Run("verifyProtocolVersion", func(t *testing.T) {
+			// protocol upgrade is async task
+			testlib.Await(t, func() bool {
+				return testlib.GetStringOccurrenceInLog(t, namespaceName, admin0,
+					fmt.Sprintf("Upgrading protocol version for database dbName=%s", opt.DbName),
+					&corev1.PodLogOptions{}) >= 1
+			}, 60*time.Second)
+
+			// fetch database effective version
+			output, err := k8s.RunKubectlAndGetOutputE(t, options.KubectlOptions, "exec", admin0, "--",
+				"nuocmd", "show", "database-versions", "--db-name", opt.DbName)
+			require.NoError(t, err, "running show database-versions failed")
+			pattern := regexp.MustCompile("effective version ID: [0-9]+, effective version: (.+),")
+			match := pattern.FindStringSubmatch(output)
+			require.NotNil(t, match, "Unable to get database effective version from output")
+
+			// effective version is string like this one 4.2|4.2.1|4.2.2; verify
+			// that the OLD_RELEASE major.minor is not in the effective version
+			// string
+			parts := strings.Split(OLD_RELEASE, ".")
+			require.GreaterOrEqual(t, len(parts), 2, "unable to get major.minor from OLD_RELEASE version")
+			require.NotContains(t, match[1], fmt.Sprintf("%s.%s", parts[0], parts[1]))
+			// verify that the new major.minor is in the effective version
+			// string
+			require.Contains(t, match[1], fmt.Sprintf("%d.%d", version.Major(), version.Minor()))
+
+			// verify that a TE has been restarted and SQL layer version is
+			// upgrade is performed
+			testlib.Await(t, func() bool {
+				return testlib.GetRegexOccurrenceInLog(t, namespaceName, admin0,
+					"Shutting down startId=[0-9]+ to finalize the database protocol upgrade",
+					&corev1.PodLogOptions{}) >= 1
+			}, 60*time.Second)
+			testlib.AwaitDatabaseUp(t, namespaceName, admin0, opt.DbName, 2)
+			output, err = testlib.RunSQL(t, namespaceName, admin0, opt.DbName,
+				"select case when version = GETEFFECTIVEPLATFORMVERSION() then 'YES' else 'NO' end as TE_RESTARTED_AFTER_MAJOR_VERSION_UPGRADE"+
+					" from system.versions where property = 'SYSTEM_TABLES_VERSION'")
+			require.NoError(t, err, "error cheking if SQL layer version is upgraded")
+			require.Contains(t, output, "YES")
+
+			// verify that the container image ID is stored in KV store so that
+			// database version is not checked again for this container image ID
+			testlib.Await(t, func() bool {
```

Comment:
It will be updated in the next resync interval after checking that there are no other available versions in the database version response and all processes are using the same release.